website/docs/reference/resource-configs/grants.md

```diff
@@ -98,32 +113,109 @@ See [configs and properties](configs-and-properties) for details.
 </TabItem>
 </Tabs>
 
-## Definition
+### Grant config inheritance
 
-You can use the `grants` field to set permissions or grants for a resource. These grants will be compiled into the `manifest.json` file complied by dbt.
+If `grants` are set for the same model in multiple places, such as in `dbt_project.yml` and in a more-specific `.sql` or `.yml` file, the default behavior is to "merge and clobber" each privilege. That is, dbt will replace the less-specific set of grantees with the more-specific set of grantees.
```

Comment:
this is so cool! utopian idea -- perhaps there should be a warning at least when there's overlap b/w model and folder grants? imagine a message like:
> You've granted `user_c` `READ` rights to the entire `mart_finance/` directory so it is redundant to grant them access specifically to the following models:
> - `fin.F22_forecast_summary`
> Would you like me to update your dbt project to represent that simplification?

idk I can imagine a world in which there's enough redundant grants to the extent that it obfuscates what's going on? 🤷🏻

Reply:
I could imagine something like that! Or maybe even: a way to recommend new folder-level configs to DRY up per-model configs. (Or to recommend folder-level reorganization...)

There's some more ideation about how configs ought to be merged (updated/clobbered/added/removed) when defined in multiple places: https://github.com/dbt-labs/dbt-core/issues/4108